packages/application-shell/src/components/authenticated/authenticated.js

```diff
@@ -1,17 +1,54 @@
+import React from 'react';
 import PropTypes from 'prop-types';
+import { Query } from 'react-apollo';
 import * as storage from '@commercetools-frontend/storage';
+import { GRAPHQL_TARGETS } from '@commercetools-frontend/constants';
 import { STORAGE_KEYS } from '../../constants';
+import AmILoggedInQuery from './authenticated.graphql';
 
-const getIsAuthenticated = () =>
+const hasCachedAuthenticationState = () =>
   storage.get(STORAGE_KEYS.IS_AUTHENTICATED) === 'true';
 
 const Authenticated = props => {
-  const isAuthenticated = getIsAuthenticated();
-  return props.children({ isAuthenticated });
+  // We attempt to see if the user was already authenticated by looking
+  // at the "cached" flag in local storage.
+  const cachedAuthenticationState = hasCachedAuthenticationState();
+  if (cachedAuthenticationState) {
+    return props.render({ isAuthenticated: true });
+  }
+
+  // ...otherwise, we ping a "secured" endpoint in the MC API to see if there is
+  // a valid access token. If we get an error, we assume that the user is not
+  // authenticated. If we don't get any error, the access token sent with the cookie
+  // is valid. We return null while the query is loading.
+  return (
+    <Query
+      query={AmILoggedInQuery}
+      variables={{ target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND }}
+      onCompleted={() => {
+        // Even though the login page might set this flag, we just make sure that
+        // we do it here as well. This will help in the future when we eventually
+        // move the auth service to itw own domain, in which case the local storage
+        // is not shared anymore.
+        storage.put(STORAGE_KEYS.IS_AUTHENTICATED, true);
+      }}
+    >
+      {({ loading, data, error }) => {
+        if (error) {
+          console.error(error);
```

Comment:
Dangling log?

Reply:
Not really, I wasn't sure what to do with that to be honest. Most likely is going to be a "unauthorized" error, which we can safely ignore.

Reply:
Should we maybe report to sentry instead?

Reply:
We will get spammed if we log on every unauthorized. Maybe we can send errors that are not unauthorized.